### ISSUE TITLE
Metric result on training dataset not outputting when is_provide_training_metric is True

### ISSUE THREAD

**Issue by ungvert** [Microsoft/LightGBM#1794]
## Environment info

Operating System: Linux Mint 18.3

CPU/GPU model: CPU

C++/Python/R version: Python

## Reproducible examples

When we pass `'is_provide_training_metric'=True`, train metric not outputted:

```python
import lightgbm as lgb
from sklearn.datasets import make_classification
from sklearn.model_selection import train_test_split


data = make_classification(random_state=0)
X_train, X_test, y_train, y_test = train_test_split(
    data[0], data[1], test_size=0.3,random_state=0)


lgb_train = lgb.Dataset(X_train,
                        label=y_train,
                        )
lgb_eval = lgb.Dataset(X_test,
                       label=y_test,
                       reference=lgb_train,
                       )
params = {"objective": "binary",
         "metric":'auc',
         'is_provide_training_metric':True}

valid_sets=(lgb_eval)
gbm = lgb.train(params,
                lgb_train,
                num_boost_round=10,
                valid_sets=valid_sets,
                early_stopping_rounds=5)


```

```

[1]	valid_0's auc: 0.875
Training until validation scores don't improve for 5 rounds.
[2]	valid_0's auc: 0.875
[3]	valid_0's auc: 0.875
[4]	valid_0's auc: 0.875
[5]	valid_0's auc: 0.893519
[6]	valid_0's auc: 0.893519
[7]	valid_0's auc: 0.893519
[8]	valid_0's auc: 0.893519
[9]	valid_0's auc: 0.893519
[10]	valid_0's auc: 0.893519
Early stopping, best iteration is:
[5]	valid_0's auc: 0.893519


gbm.best_score


defaultdict(dict, {'valid_0': {'auc': 0.8935185185185185}})

```

When we pass the train as part of eval set, the train metrics outputted, even when `'is_provide_training_metric':False`:

``` python

params = {"objective": "binary",
         "metric":'auc',
         'is_provide_training_metric':False}

valid_sets=(lgb_eval,lgb_train)
gbm = lgb.train(params,
                lgb_train,
                num_boost_round=10,
                valid_sets=valid_sets,
                early_stopping_rounds=5)
```

```
[1]	training's auc: 0.929276	valid_0's auc: 0.875
Training until validation scores don't improve for 5 rounds.
[2]	training's auc: 0.929276	valid_0's auc: 0.875
[3]	training's auc: 0.929276	valid_0's auc: 0.875
[4]	training's auc: 0.929276	valid_0's auc: 0.875
[5]	training's auc: 0.942023	valid_0's auc: 0.893519
[6]	training's auc: 0.942023	valid_0's auc: 0.893519
[7]	training's auc: 0.972451	valid_0's auc: 0.893519
[8]	training's auc: 0.972451	valid_0's auc: 0.893519
[9]	training's auc: 0.972451	valid_0's auc: 0.893519
[10]	training's auc: 0.972451	valid_0's auc: 0.893519
Did not meet early stopping. Best iteration is:
[7]	training's auc: 0.972451	valid_0's auc: 0.893519


gbm.best_score

defaultdict(dict,
            {'training': {'auc': 0.9724506578947368},
             'valid_0': {'auc': 0.8935185185185185}})
```

Is this intended or I missing something?
